internal/dataplane/kong_client.go

```diff
@@ -311,13 +311,9 @@ func (c *KongClient) Update(ctx context.Context) error {
 	}
 
 	// parse the Kubernetes objects from the storer into Kong configuration
-	kongstate, err := p.Build()
-	if err != nil {
-		c.prometheusMetrics.TranslationCount.With(prometheus.Labels{
-			metrics.SuccessKey: metrics.SuccessFalse,
-		}).Inc()
-		return err
-	}
+	kongstate := p.Build()
+	// todo: does it still make sense to report TranslationCount when Build no longer returns an error?
```

Comment:
It makes perfect sense to me. Although the success label has little sense now we can keep it for backwards compatibility. It will do no harm since we're actually not going to fail now, are we?

Reply:
I'll leave this todo and will try to address it in the design doc I'm working on. Once we will have a mechanism to tell if any object was invalid and skipped during the `Build` call, we could inspect that and report failure case potentially, treating `Build` to be successful only when no object gets skipped.